evm/vm/base.py

```diff
@@ -234,40 +471,31 @@ def get_state_class(cls):
 
         return cls._state_class
 
-    def get_state(self):
+    def get_state(self, chaindb=None, block_header=None, prev_headers=None):
         """Return state object
         """
+        if chaindb is None:
+            chaindb = self.chaindb
+        if block_header is None:  # TODO: remove
```

Comment:
Can you maybe explain (in the comment) why we want to remove this?

Reply:
Ah sorry, I was thinking about if I can store `block_header` in `prev_headers` in some cases, but it turns out because `block_header` is more like a container object in `VMState`, it's more reasonable to leave `block_header` alone now. I'll remove this TODO tag.

Furthermore, there's another TODO issue I mention in this PR and #256, for sharding clients, we need to decouple `VMState.block_header` the `VMState` properties (`blockhash, timestamp, block_number, difficulty, and gas_limit)`, in that case, the parameters of `VMState.__init__` may become something like:
```python
def __init__(self, chaindb, state_root, block_header, prev_headers, receipts=[])
```
Thanks!